qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,121 @@
+    virtio_drivers = []
+    drivers = {}
+    need_test_driver = params.objects("tested_drivers")
+    get_disk_letter = params.get("get_disk_letter_cmd")
+
+    def cat_filter(driver):
+        pattern = r'(\w:)\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        ret = re.match(pattern, driver, re.I)
+        disk_name = ret.group(1)
+        os_name = ret.group(3)
+
+        if disk_name == "A:":
+            sub_os_name = os_dirs[0]
+            plat_name = ret.group(2)
+            driver_dir = ret.group(4)
+        else:
+            sub_os_name = os_dirs[1]
+            driver_dir = ret.group(2)
+            plat_name = ret.group(4)
+
+        if sub_os_name in os_name:
+            if arch_dir in plat_name:
+                if driver_dir in need_test_driver:
+                    return True
+
+    def match_cat(cat_element):
+        pattern = r'\w:\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        match_ret = re.match(pattern, cat_element)
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (cat_element, x), driver_sys)
+        cat_inf = map(lambda x: (cat_element, x), driver_inf)
+        cat_wdf = map(lambda x: (cat_element, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        disk_result = session.cmd_output(get_disk_letter)
+        disk_letter = re.search(r'(\w:)\s', disk_result, re.M).group(1)
+        drive_list.append(disk_letter)
+
+        for drive in drive_list:
+            for type in ['.cat', '.sys', '.inf', 'Wdf']:
                 driver = session.cmd_output(list_files_cmd % (drive, type)).\
                     splitlines()[1:-1]
                 drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
+
+            cat_list = filter(cat_filter, drivers['.cat'])
+            map(match_cat, cat_list)
+
+            for driver_file in virtio_drivers:
+                test_cmd = signtool_cmd % (driver_file[0], driver_file[1])
+                test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
                     raise error.TestFail(msg)
```

Comment:
peixiu, pls change to test.fail() Incidentally

Reply:
has updated, thanks~